test/tbb/test_arena_constraints.cpp

```diff
@@ -212,3 +212,10 @@ TEST_CASE("Test concurrency getters output for constraints with custom concurren
     c.set_max_threads_per_core(1);
     check_concurrency_level(c);
 }
+
+//! Testing constraints_threads_per_core() reserved entry point
+//! \brief \ref interface \ref error_guessing
+TEST_CASE("Testing constraints_threads_per_core() reserved entry point") {
+    tbb::task_arena::constraints c{};
+    REQUIRE(tbb::detail::r1::constraints_threads_per_core(c) == tbb::task_arena::automatic);
```

Comment:
Do we expect that this behavior will be preserved in future that `constraints_threads_per_core` will return `automatic` for the default constructed constraints?

Reply:
No, I think it will return something like a maximal number of threads per core for the given constraints.

Reply:
It seems that the test will not be backward compatible in that case. Consider just calling this entry point.